src/pmdas/amdgpu/amdgpu.c

```diff
@@ -0,0 +1,547 @@
+/*
+ * Copyright (c) 2024 Red Hat.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * for more details.
+ */
+#include "pmapi.h"
+#include "pmda.h"
+
+#include "domain.h"
+#include "libpcp.h"
+
+#include "localdrm.h"
+#include <stdint.h>
+
+/* InDom table (set of graphics cards, set of processes+devices) */
+enum { GCARD_INDOM = 0 };
+pmdaIndom indomtab[] = {
+    {GCARD_INDOM, 0, NULL},
+};
+
+/* List of metric item numbers - increasing from zero, no holes.
+ * Double check against `pmns` definition file.
+ */
+enum {
+  AMDGPU_NUMCARDS = 0,
+  AMDGPU_CARDNAME,
+  AMDGPU_MEMUSED,
+  AMDGPU_MEMTOTAL,
+  AMDGPU_MEMFREE,
+  AMDGPU_SAMPLES,
+  AMDGPU_MEMUSED_ACCUM,
+  AMDGPU_GPU_CLOCK,
+  AMDGPU_GPU_CLOCK_MAX,
+  AMDGPU_MEMORY_CLOCK,
+  AMDGPU_MEMORY_CLOCK_MAX,
+  AMDGPU_TEMPERATURE,
+  AMDGPU_GPU_LOAD,
+  AMDGPU_GPU_AVG_PWR,
+
+  AMDGPU_METRIC_COUNT
+};
+
+/* Table of metrics exported by this PMDA */
+static pmdaMetric metrictab[] = {
+    {NULL,
+     {PMDA_PMID(0, AMDGPU_NUMCARDS), PM_TYPE_U32, PM_INDOM_NULL, PM_SEM_INSTANT,
+      PMDA_PMUNITS(0, 0, 0, 0, 0, 0)}},
+    {NULL,
+     {PMDA_PMID(0, AMDGPU_CARDNAME), PM_TYPE_STRING, GCARD_INDOM, PM_SEM_DISCRETE,
+      PMDA_PMUNITS(0, 0, 0, 0, 0, 0)}},
+    {NULL,
+     {PMDA_PMID(0, AMDGPU_MEMUSED), PM_TYPE_U64, GCARD_INDOM, PM_SEM_INSTANT,
+      PMDA_PMUNITS(1, 0, 0, PM_SPACE_BYTE, 0, 0)}},
+    {NULL,
+     {PMDA_PMID(0, AMDGPU_MEMTOTAL), PM_TYPE_U64, GCARD_INDOM, PM_SEM_DISCRETE,
+      PMDA_PMUNITS(1, 0, 0, PM_SPACE_BYTE, 0, 0)}},
+    {NULL,
+     {PMDA_PMID(0, AMDGPU_MEMFREE), PM_TYPE_U64, GCARD_INDOM, PM_SEM_INSTANT,
+      PMDA_PMUNITS(1, 0, 0, PM_SPACE_BYTE, 0, 0)}},
+    {NULL,
+     {PMDA_PMID(0, AMDGPU_SAMPLES), PM_TYPE_U64, GCARD_INDOM, PM_SEM_COUNTER,
+      PMDA_PMUNITS(0, 0, 0, 0, 0, 0)}},
```

Comment:
Units for a basic counter are usually like:

PMDA_PMUNITS(0,0,1,0,0,PM_COUNT_ONE)